project/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -97,37 +97,37 @@ public void requestHttpAndBuildJsonResponseFromCivicInformation_returnResponseSt
     when(httpClient.execute(anyObject(), argumentCaptor.capture()))
         .thenReturn(RELEVANT_ADDRESS_ELECTION_RESPONSE);
     String responseBody =
-        dataServlet.requestHttpAndBuildJsonResponseFromCivicInformation(httpClient, httpGet);
+        dataServlet.requestHttpAndBuildCivicInfoResponse(httpClient, httpGet);
+    httpClient.close();
     assertThat(responseBody).isEqualTo(RELEVANT_ADDRESS_ELECTION_RESPONSE);
   }
 
   @Test
   public void parseResponseForRelevancy_relevantElection() {
     // Parse JSON-formatted String {@code RELEVANT_ADDRESS_ELECTION_RESPONSE} and see that it
-    // indicates the election is relevant to the address according to the Civic Information
-    // API response structure.
-    assertThat(dataServlet.parseResponseForRelevancy(RELEVANT_ADDRESS_ELECTION_RESPONSE))
-        .isTrue();
+    // indicates the election is relevant to the address and the address is both specific and
+    // counts as a residential address, according to the Civic Information API response structure.
+    dataServlet.parseResponseForRelevancy(RELEVANT_ADDRESS_ELECTION_RESPONSE);
+    assertThat(dataServlet.isAddressRelevantButNonspecificOrNonresidential).isFalse();
   }
 
   @Test
   public void parseResponseForRelevancy_relevantNonspecificElection() {
     // Parse JSON-formatted String {@code RELEVANT_NONSPECIFIC_ADDRESS_ELECTION_RESPONSE} and see
     // that it,indicates the election is probably relevant to the address according to the Civic
     // Information API response structure, but the address is nonspecific.
-    dataServlet.isAddressRelevantButNonspecific = false;
-    assertThat(
-        dataServlet.parseResponseForRelevancy(RELEVANT_NONSPECIFIC_ADDRESS_ELECTION_RESPONSE))
-        .isTrue();
-    assertThat(dataServlet.isAddressRelevantButNonspecific).isTrue();
+    dataServlet.isAddressRelevantButNonspecificOrNonresidential = false;
+    dataServlet.parseResponseForRelevancy(RELEVANT_NONSPECIFIC_ADDRESS_ELECTION_RESPONSE);
+    assertThat(dataServlet.isAddressRelevantButNonspecificOrNonresidential).isTrue();
   }
 
   @Test
   public void parseResponseForRelevancy_irrelevantElection() {
     // Parse JSON-formatted String {@code IRRELEVANT_ADDRESS_ELECTION_RESPONSE} and see that it
     // indicates the election is irrelevant to the address according to the Civic Information
     // API response structure.
-    assertThat(dataServlet.parseResponseForRelevancy(IRRELEVANT_ADDRESS_ELECTION_RESPONSE))
-        .isFalse();
+    dataServlet.isAddressRelevantButNonspecificOrNonresidential = false;
+    dataServlet.parseResponseForRelevancy(IRRELEVANT_ADDRESS_ELECTION_RESPONSE);
+    assertThat(dataServlet.isAddressRelevantButNonspecificOrNonresidential).isTrue();
```

Comment:
Why woudl this be true for a non-relevant address?

Reply:
The test isn't testing what it should. I'll rewrite some tests!